be/src/storage/rowset_update_state.cpp

```diff
@@ -103,6 +109,130 @@ Status RowsetUpdateState::_do_load(Rowset* rowset) {
     return Status::OK();
 }
 
+struct RowidSortEntry {
+    uint32_t rowid;
+    uint32_t idx;
+    RowidSortEntry(uint32_t rowid, uint32_t idx) : rowid(rowid), idx(idx) {}
+    bool operator<(const RowidSortEntry& rhs) const { return rowid < rhs.rowid; }
+};
+
+static void plan_read_by_rssid(const vector<uint64_t>& rowids, size_t* num_default,
+                               std::map<uint32_t, std::vector<uint32_t>>* rowids_by_rssid, vector<uint32_t>* pidxes) {
+    uint32_t n = rowids.size();
+    phmap::node_hash_map<uint32_t, vector<RowidSortEntry>> sort_entry_by_rssid;
+    std::vector<uint32_t> defaults;
+    for (uint32_t i = 0; i < n; i++) {
+        uint64_t v = rowids[i];
+        uint32_t rssid = v >> 32;
+        if (rssid == (uint32_t)-1) {
+            defaults.push_back(i);
+        } else {
+            uint32_t rowid = v & 0xffffffffUL;
+            sort_entry_by_rssid[rssid].emplace_back(rowid, i);
+        }
+    }
+    *num_default = defaults.size();
+    bool has_default = num_default > 0;
+    std::vector<uint32_t>& idxes = *pidxes;
```

Comment:
can use pointer directly, or pass a non-const reference and use it directly(google coding style do not restrict it anymore), instead of dereferencing a pointer.